examples/larger-examples/datalog/subtyping.flix

```diff
@@ -0,0 +1,176 @@
+mod Subtyping {
+
+    type alias Program = {
+        classes = Vector[String],
+        finalClasses = Vector[String],
+        interfaces = Vector[String],
+        implementations = Vector[(String, String)],
+        classExtends = Vector[(String, String)],
+        interfaceExtends = Vector[(String, String)],
+        methods = Vector[(String, String)]
+    }
+
+    def subtypes(program: Program): #{ Subtype(String, String), AvailableMethod(String, String), NonSupertype(String), ExtendsFinalClass(String, String) | r } =
+        let rules = #{
+            Subtype(x, x)       :- Class(x).
+            Subtype(x, x)       :- Interface(x).
+            Subtype(sub, super) :- Class(sub), Interface(super), Implementation(sub, super).
+            Subtype(sub, super) :- Class(sub), Class(super), ClassExtends(sub, super).
+            Subtype(sub, super) :- Interface(sub), Interface(super), InterfaceExtends(sub, super).
+            Subtype(sub, super) :- Subtype(sub, a), Subtype(a, super).
+
+            AvailableMethod(class, method) :- Method(t, method), Subtype(class, t).
+            NonSupertype(class)            :- Class(class), Class(otherClass), not Subtype(otherClass, class), if class != otherClass.
+            ExtendsFinalClass(sub, super)  :- FinalClass(super), Subtype(sub, super), if sub != super.
```

Comment:
I want to move this rule into a separate method. Then that method can be given finalClasses as an argument.